### PR TITLE
Check XML doc parameter names for FSharp.Core via opt-in checking

### PR DESCRIPTION
**Pull request by dsyme**
1. Fixes mistakes in FSharp.Core XML docs

2. Fix mistakes in FCS docs 

3. Implements the opt-in XML document checking `/warnon:3390` and applies it to FSharp.Core, 

   When `/warnon:3390` is enabled (e.g. for FCS and FSharp.Core):

   * checks `<param name="xyz">` and `<paramref name="xyz">` it checks that `xyz`  is a valid parameter name for the thing being documents

   * checks that if any parameter is given docs, then all are given docs

4. Adds the ability to get the elaborated (e.g. with summary tags) XML from the FCS API, which we should always have allowed. This will become more important if/when we support cross-reference `cref` elaboration in XML docs.

5. Fixes a bug where you couldn't add documentation to an implicit constructor, see [this stack overflow question](https://stackoverflow.com/questions/15121814/documenting-f-code).  

Aside: This PR also contains the beginning of code for cross-reference checking and elaboration, allowing the use of "short name" forms for cross references, including rename-refactor on these.  However that code is not enabled.

### CODE REVIEW

**Comment by dsyme** [comment]
This is ready (once green - having some niggling issues)

**Comment by dsyme** [comment]
This is now ready for review

**Comment by dsyme** [comment]
@cartermp I think it makes sense to document 

* `/warnon:1182` - unused variables
* `/warnon:3390` - check XML docs for validity and completeness

in the Microsoft F# docs.  I was going to put something under "docs" but it's likely better in the actual guide

**Comment by dsyme** [comment]
Side note: I wanted to add separate  warnings for 

* give warning if elements do not have documentation

* give warning if an element doesn't have full parameter information

However there are some issues with that.  Notably 

1. It forces people to provide documentation for implicit constructors (or else have a signature file).  This is awkward, e.g. 

   ```
   /// The type docs
   type C
       /// The implicit constructor docs
       (x: int, y: int) = ...
   ```

   I don't really want a feature that will force this on people.  We should probably allow a special section in the docs for the type e.g.

   ```
   /// <summary>The type docs</summary>
   /// <ctor><summary>The implicit constructor docs</summary></ctor>
   type C(x: int, y: int) = ...
   ```


2. It is also common to have incomplete docs (e,g, no parameter docs)  in an implementation but have complete docs in a signature.  So enforcement of completeness should only be done on the signature docs if a signature is present.

File: src/fsharp/xlf/FSComp.txt.ja.xlf [hunk 1]
```diff
@@ -2,7559 +2,7594 @@
 <xliff xmlns="urn:oasis:names:tc:xliff:document:1.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="1.2" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 xliff-core-1.2-transitional.xsd">
   <file datatype="xml" source-language="en" target-language="ja" original="../FSComp.resx">
     <body>
-      <trans-unit id="chkFeatureNotLanguageSupported">
```
Comment (cartermp) [inline src/fsharp/xlf/FSComp.txt.ja.xlf:5]: How did this enormous diff occur?

**Review by KevinRansom — APPROVED** [review]

File: src/fsharp/xlf/FSComp.txt.ja.xlf (hunk 1 above)
Comment (dsyme) [reply src/fsharp/xlf/FSComp.txt.ja.xlf:5]: Not sure

**Comment by dsyme** [comment]
@KevinRansom Thank you, it's great to have this in

**Comment by dsyme** [comment]
Relevant documentation PR: https://github.com/dotnet/docs/pull/20645

**Pull request merged**